core/tests/performance_tests/string_classifier_performance_test.py

```diff
@@ -0,0 +1,143 @@
+        classifier: dict. The model generated by function train.
+    """
+    def __init__(self):
+        """ Initializes StringClassifierBenchmark by parsing the data
+        file containing samples
+        """
+        yaml_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                 '../data/string_classifier_test.yaml')
+        doc_to_label = {}
+        with open(yaml_path, 'r') as yaml_file:
+            yaml_dict = yaml.load(yaml_file)
+            interactions = yaml_dict['states']['Home']['interaction']
+            for interaction in interactions['answer_groups'][1:]:
+                label = interaction['outcome']['feedback'][0]
+                for rule in interaction['rule_specs']:
+                    if "inputs" in rule and 'training_data' in rule['inputs']:
+                        for doc in rule['inputs']['training_data']:
+                            if doc not in doc_to_label:
+                                doc_to_label[doc] = []
+                            doc_to_label[doc].append(label)
+        self.examples = [[doc, doc_to_label[doc]] for doc in doc_to_label]
+        self.predict_docs = [doc[0] for doc in self.examples]
+        self.classifier = None
+
+    @benchmark
+    def train(self, num):
+        """ Trains a model with num samples.
+
+        Args:
+            num: int. The number of samples to run with.
+
+        Returns:
+            dict. The resulting dict representing the model.
+        """
+        string_classifier = StringClassifier()
+        string_classifier.load_examples(self.examples[:num])
+        classifier_dict = string_classifier.to_dict()
+        return classifier_dict
+
+    def benchmark_on_train(self):
+        """ Conduct benchmarking on model training at a pace of adding
+        100 samples each time
+        """
+        for num in xrange(100, len(self.examples), 100):
+            self.train(num)
+
+    @benchmark
+    def predict(self, num):
+        """ Predicts the label for num samples with self.classifier.
+
+        Args:
+            num: int. The number of samples to run with.
+        """
+        if not self.classifier:
+            raise Exception('No classifier found')
+        string_classifier = StringClassifier()
+        string_classifier.from_dict(self.classifier)
+        doc_ids = string_classifier.add_docs_for_predicting(
+            self.predict_docs[:num])
+        for i in xrange(len(doc_ids)):
+            string_classifier.predict_label_for_doc(doc_ids[i])
+
+    def benchmark_on_predict(self):
+        """ Conduct benchmarking on predicting with self.classifier
+        at a pace of adding 100 samples each time
+        """
+        self.classifier = self.train(len(self.examples))
+        for num in xrange(100, len(self.predict_docs), 100):
+            self.predict(num)
+
+
+if __name__ == '__main__':
+    benchmark = StringClassifierBenchmarker() #pylint: disable=invalid-name
```

Comment:
Add a "def main()" function and pull these three statements out into it, and call main() here -- then I don't think you'll have the invalid name issue. It's more conventional to do that, too, I think.